client/jetpack-onboarding/main.jsx

```diff
@@ -61,8 +64,10 @@ export default connect( ( state, { siteSlug } ) => {
 	] );
 
 	return {
-		siteId: getUnconnectedSiteIdBySlug( state, siteSlug ),
+		jpUser: get( state.jetpackOnboarding.credentials, [ siteId, 'userEmail' ], null ),
+		siteId,
 		siteSlug,
 		steps,
+		token: get( state.jetpackOnboarding.credentials, [ siteId, 'token' ], null ),
 	};
```

Comment:
Alternatively, we could use `connect()`'s second and third arguments, [`mapDispatchToProps` and `mergeProps`](https://github.com/reactjs/react-redux/blob/master/docs/api.md#connectmapstatetoprops-mapdispatchtoprops-mergeprops-options) (rather advanced usage of `connect()` :smile:) to pass a pre-bound `recordTracksEvent` function prop. Something like

```
( state, { siteSlug } ) => { /* ... */ },
{ recordTracksEvent },
( { jpUser, token, ...otherStateProps }, { recordTracksEvent } ) => ( {
	...otherStateProps,
	recordTracksEvent: event => recordTracksEvent( event, {
		blog_id: stateProps.token,
		site_id_type: 'jpo',
		user_id_type: stateProps.jpUser,
	} )
} )
```

If this works, it will allow us to call `recordTracksEvents` from individual steps like so:

```js
this.props.recordTracksEvent( 'calypso_jpo_contact_form_clicked' );
```

Reply:
What Bernie said sounds like a good approach. We'll want to avoid defining this in all step components (we want to try to keep steps as minimal as possible), so it would probably make sense to define it in one place, and reuse it everywhere. To solve this, we can define it in `main` and pass it down to all steps. 

Alternative option is to create an action creator that will be used for all those cases. 

Yet another option is to create a high order component to provide that action to all steps.